### ISSUE TITLE
Promotion codes aren't really case sensitive

### ISSUE THREAD

**Issue by nirebu** [solidusio/solidus#4328]
Promotion codes have this validation:

```rb
# app/models/spree/promotion_codes.rb

validates :value, presence: true, uniqueness: { allow_blank: true, case_sensitive: true }
```

and this callback:

```rb
before_save :normalize_code

private

def normalize_code
  self.value = value.downcase.strip
end
```

This makes Rails check for an existing record with the same **case sensitive** code when creating a promotion code (e.g. `code40`), and if we have the same code with different casing (e.g. `CODE40`), we won't find it. If this happens, the record would be considered valid and an `INSERT` is initiated, only to be refused by the DB where we have a unique index on the Promotion Codes value column, thus generating a 500 in the admin panel.

**Solidus Version**:
3.1.4

**To Reproduce**
- Create a promotion with `CODE40` as the promotion code
- Create a promotion with `code40` as the promotion code

**Current behavior**
DB level exception is raised

**Expected behavior**
A validation error is raised

**Additional context**
Given the normalization pass before saving, I'm assuming that the intended behaviour would be to have **case insensitive** promotion codes, to help in URL sharing and similar applications. My suggestion would be to change the `case_sensitive: true` to `false` to leverage this validation.

**Comment by kennyadsl** [comment]
Hey @nirebu! Thanks for submitting the issue, which seems to make sense. Do you think we'll have any backward-compatibility issues with the proposed change?

**Comment by kennyadsl** [comment]
@nirebu I think this may be related: https://github.com/solidusio/solidus/pull/4228. This is on master and not yet released but we plan to have a Solidus 3.1.x release this this fix. Let me know if this solves and we can craft the release soon.

**Comment by nirebu** [comment]
@kennyadsl absolutely! That's exactly the issue I've encountered. I must admit that explanation is way better than mine, I believe we can close this one 🚀
